code/obj/zoldorf.dm

```diff
@@ -85,6 +40,114 @@ var/global/list/datum/zoldorfitem/zoldorf_items = list()
 	New()
 		. = ..()
 		START_TRACKING
+		for(var/product in childrentypesof(/datum/zoldorfitem/soul))
+			src.soul_items += new product
+		for(var/product in childrentypesof(/datum/zoldorfitem/credit))
+			src.credit_items += new product
```

Comment:
Not verified this works, but possibly (takes advantage of abstract annotations):
```suggestion
		for(var/product in concrete_typesof(/datum/zoldorfitem))
			src.soul_items += new product
```